mitodl/mit-open
# Database index on FeedEventDetail.event_datetime

### What are the relevant tickets?
- https://github.com/mitodl/hq/issues/4883

### Description (What does it do?)
- Adds a database index on `news_events.models.FeedEventDetail.event_datetime` 


### How can this be tested?
- Not strictly necessary but setting `DEBUG=False` in your .env file will likely speed up response times.
- Also set `SILKY_PYTHON_PROFILER=True` in your backend env file.
- If you don't have any news/events in the database yet, you can run `./manage.py backpopulate_news_events`.  You can also generate fake ones in a shell to get the numbers up (the command will likely import only a dozen or two):
  ```
  from news_events.factories import *
  source = FeedSourceFactory.create(is_event_type=True)
  FeedItemFactory.create_batch(100, source=source, is_event=True)
  ```
- In postman or equivalent, send multiple requests to http://api.open.odl.local:8063/api/v0/news_events/?feed_type=events
- Go to http://api.open.odl.local:8063/silk/ and check that the # of queries and response time seem acceptable.
- The home page should still show events ordered by event date (soonest first).

![Screenshot 2024-07-23 at 8 52 41 AM](https://github.com/user-attachments/assets/1b10feff-7be7-4d6d-bf84-1e7dd021cd77)
